include/parallel/parallel_implementation.h

```diff
@@ -2052,6 +2049,29 @@ inline status Communicator::probe (const unsigned int src_processor_id,
   return stat;
 }
 
+template<typename T>
+inline Status Communicator::packed_range_probe (const unsigned int src_processor_id,
+                                                const MessageTag & tag,
+                                                bool & flag) const
+{
+  START_LOG("packed_range_probe()", "Parallel");
+
+  Status stat((StandardType<typename Packing<T>::buffer_type>()));
+
+  int int_flag;
+
+  libmesh_call_mpi(MPI_Iprobe(src_processor_id,
+                              tag.value(),
+                              this->get(),
+                              &int_flag,
+                              stat.get()));
+
+  flag = int_flag;
+
+  STOP_LOG("packed_range_probe()", "Parallel");
+
+  return stat;
+}
 
```

Comment:
No way to merge it with the existing probe... that one is blocking and doesn't set a flag.

You are right about the datatype.  Either way with the datatype I just wanted to make sure it was inferred from the `Packing` buffer type.  This was the first way I thought to do it - but I like your idea of doing it in `nonblocking_send_packed_range()` better.  Don't know why I didn't think of that ;-)

I could make it look just like the other call - except add in the flag.

That said, I kind of think the existing `probe()` is a bit inconsistent.  Shouldn't we make `Communicator::probe()` return our own datatype (`Status`)?  Right now it returns a plain `MPI_Status` (do we even know that it's ok to return one of those by copy?).

Personally I would really like the new, non-blocking `probe()` to return a `bool` that is governed by the `flag` (and it wouldn't take a `flag` argument at all).

That way code could look like:

```
if (comm.nonblocking_probe(0, my_tag, my_status))
  comm.nonblocking_receive_packed_range(my_context, &my_type, my_status);
```

Then `nonblocking_receive_packed_range` will use `status.source()` and `status.tag()` implicitly.  The nice thing is... it will tie these two functions together even more closely... you definitely won't be able to use `nonblocking_receive_packed_range()` without first getting a status.

What do you think?

BTW: thanks for the suggestions.  I definitely want to get this right.  The MPI stuff in `Parallel` is really awesome and I want this to fit in the right way!